scripts/testTimezones.js

```diff
@@ -0,0 +1,118 @@
+// command line:
+//
+// ```bash
+// tzutil /s "Eastern Standard Time"
+// ```
+// (replacing `Eastern Standard Time`) with your timezone)
+
+const isWindows = os.platform() === 'win32';
+const currentSystemTimezone = () => Intl.DateTimeFormat().resolvedOptions().timeZone;
+const startingTimezone = currentSystemTimezone();
+
+const timezones = ['America/Chicago', 'UTC', 'Australia/Sydney', startingTimezone];
+
+const run = async () => {
+  console.log(`⏲  Initial timezone: ${startingTimezone}`);
+
+  let failures = 0;
+  for (const timezone of timezones) {
+    setTZ(timezone);
+    await pause(); // sometimes takes a few ticks to register
+
+    console.log('');
+    console.log('---------------------------');
+    console.log(`⏲  Timezone changed to ${timezone}`);
+    console.log('');
+    try {
+      await runTest(timezone);
+      console.log(`✅ Tests passed for timezone ${timezone}`);
+    } catch (err) {
+      console.error(`❌ Tests failed for timezone ${timezone}`);
+      failures++;
+    }
+    await pause();
+  }
+
+  console.log('');
+  console.log('---------------------------');
+  if (failures === 0) console.log(`✅ Tests passed for all ${timezones.length} timezones`);
+  else console.error(`❌ Tests failed for ${failures} out of ${timezones.length} timezones`);
+
+  console.log(`⏲  Timezone reset to ${currentSystemTimezone()}`);
+
+  return failures ? 1 : 0;
+};
+
+const runTest = async () => {
+  return new Promise((resolve, reject) => {
+    const results = [];
+    const pipeOutputToConsole = { stdio: [process.stdin, process.stdout, process.stderr] };
+    const testWorker = spawn('node', ['./scripts/testTimezones_worker.js'], pipeOutputToConsole);
+    testWorker.on('close', code => {
+      if (code === 0) resolve(results);
+      else reject(results);
+    });
+  });
+};
+
+// Utilities to change & restore the system timezone
+
+const setTZ = inputTZ => {
+  const tz = lookupTZ(inputTZ);
+  if (isWindows) {
+    // on windows we use the command-line `tzutil` utility
+    execSync(`tzutil /s "${tz}"`);
+    process.on('exit', restoreTZ); // clean up when done
+    process.on('SIGINT', () => {
+      restoreTZ();
+      process.exit(2);
+    }); // clean up when interrupted with ctrl-C
+  } else {
+    // on linux etc. we can just set this environment variable
+    process.env.TZ = tz;
+  }
+};
+
+const restoreTZ = () => {
+  if (startingTimezone && startingTimezone !== currentSystemTimezone()) {
+    setTZ(startingTimezone);
+  }
+};
+
+const lookupTZ = inputTZ => {
+  if (inputTZ === 'UTC') return inputTZ;
+  const tz = isWindows
+    ? ianaWin.findWindows(inputTZ) //
+    : ianaWin.findOneIana(inputTZ) || inputTZ;
+  if (!tz) throw new Error(`The timezone '${inputTZ}' does not exist. Please provide a valid Windows or IANA time.`);
+  return tz;
+};
+
+const pause = (t = 500) => new Promise(ok => setTimeout(() => ok(), t));
+
+// Execution
+
+run().then(result => {
+  process.exit(result);
+});
```

Comment:
Could we add a catch and reset the timezone in case we changed it?

Reply:
see next comment